filecoin-proofs/src/api/internal.rs

```diff
@@ -476,7 +500,9 @@ pub fn seal<T: Into<PathBuf> + AsRef<Path>>(
         None,
     )?;
 
-    write_data(out_path, &data)?;
+    // If we succeeded in replicating, flush the data and protect output from being cleaned up.
+    data.flush()?;
+    cleanup.success = true;
```

Comment:
Interesting idea. I don't use the `Drop` trait enough.

Reply:
Yeah, it's the Rusty way to implement RAII.

Reply:
(The mopre fleshed-out version of this work would move initialization/allocation into the struct too.)